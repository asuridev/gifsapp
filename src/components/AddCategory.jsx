import { useState } from "react";

export const AddCategory = ({ onAddCategory }) => {
  const [value, setValue] = useState("");

  const handleChange = (e) => {
    setValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    onAddCategory(value.trim());
    setValue("");
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        value={value}
        onChange={handleChange}
        type="text"
        placeholder="Buscar Gifs"
      />
    </form>
  );
};
