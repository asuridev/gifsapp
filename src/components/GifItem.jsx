export const GifItem = ({ data }) => {
  const { title, images } = data;
  const {
    downsized_medium: { url },
  } = images;

  return (
    <div className="card">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
};
