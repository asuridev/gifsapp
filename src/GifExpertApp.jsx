import { useState } from "react";
import { AddCategory, GifGrid } from "./components";


export const GifExpertApp = () => {
  const [categories, setCategories] = useState([]);

  const onAddCategory = (category) => {
    if (category.length <= 1) return;
    const categoriesUpperCase = categories.map((category) =>
      category.toUpperCase()
    );
    if (categoriesUpperCase.includes(category.toUpperCase())) return;
    setCategories([category, ...categories]);
  };

  return (
    <>
      <h1>GifExpertApp</h1>
      <AddCategory onAddCategory={onAddCategory} />

      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </>
  );
};
