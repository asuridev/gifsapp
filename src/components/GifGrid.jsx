import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "../components";

export const GifGrid = ({ category }) => {
  const { data, isLoading } = useFetchGifs(category);

  if (isLoading) {
    return <h2>Cargando...</h2>;
  }
  return (
    <div>
      <h3>{category}</h3>
      <div className="card-grid">
        {data.map((item) => (
          <GifItem
            key={item.id}
            title={item.title}
            url={item.images.downsized_medium.url}
          />
        ))}
      </div>
    </div>
  );
};
