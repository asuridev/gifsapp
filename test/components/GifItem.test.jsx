import { render } from "@testing-library/react";
import { GifItem } from "../../src/components";

describe("validando el componente <GifItem />", () => {
  test("debe hacer match con el snapshot", () => {
    const { container } = render(<GifItem title="hola mundo" url="www.jgjgj.jpg" />);
    expect(container).toMatchSnapshot();
  });
});
