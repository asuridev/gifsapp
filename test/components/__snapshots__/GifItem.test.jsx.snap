// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validando el componente <GifItem /> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="hola mundo"
      src="www.jgjgj.jpg"
    />
    <p>
      hola mundo
    </p>
  </div>
</div>
`;
